src/main/java/io/kokuwa/maven/helm/UploadMojo.java

```diff
@@ -167,6 +203,39 @@ private void uploadSingle(Path chart) throws MojoExecutionException, IOException
 		connection.disconnect();
 	}
 
+	private boolean verifyUpload(Path chartPath) throws MojoExecutionException {
+		ObjectMapper MAPPER = new YAMLMapper()
+				.configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
+		String chartName;
+		try {
+			chartName = MAPPER.readValue(chartPath.toFile(), HelmChart.class).getName();
+		} catch (IOException e) {
+			throw new MojoExecutionException("Unable to read chart from " + chartPath, e);
+		}
+
+		long startTime = System.currentTimeMillis();
+		long timeoutMillis = uploadVerificationTimeout * 1000;
+		boolean verificationSuccess = false;
+
+		while (System.currentTimeMillis() - startTime < timeoutMillis && !verificationSuccess) {
+			try {
+				helm()
+					.arguments("show", "chart", chartName,
+						"--version", getChartVersion(), "--repo", getHelmUploadUrl())
+					.execute("show chart failed");
+				verificationSuccess = true;
+			} catch (Exception e) {
+				getLog().info("Upload verification failed, retrying...");
+				try {
+					Thread.sleep(10000);
```

Comment:
S: 10 seconds seems long. I'd drop it to 1 second.  There are some cool backoff algorithms that extend the sleep longer and longer the more the request fails, but in lieu of that 1s shouldn't overload any web service.